plugins/coverbrowser.koplugin/bookinfomanager.lua

```diff
@@ -493,36 +493,10 @@ function BookInfoManager:extractBookInfo(filepath, cover_specs)
         if loaded then
             dbrow.pages = pages
             local props = document:getProps()
-            if next(props) then -- there's at least one item
-                dbrow.has_meta = 'Y'
+            dbrow.has_meta = 'Y' -- title always exists
```

Comment:
This `has_meta` is only used to enable/disable the Ignore metadata button.
But the presence of absence of a real title metadata used to make the look of a mosaic item different:
![image](https://github.com/koreader/koreader/assets/24273478/ff133567-435b-4628-93fc-03feac748d2b)
On the left, before this PR - on the right, with this PR.
I somehow like this mini feature, making it clear which book has metadata.

I'm fine with the deduplication - but may be you should add an other bool property in getProps(), `synthetized_title = true/false` (or a better name :)), so the bits of code that did differently whether title exists or not can continue doing so by checking this property (dunno if others than Mosaic items do).

And/or also add a `has_meta = true/false` in getProps() - so even if you synthetized title (and may be other ones), other code can know there actually was no usable metadata in that document - even if using synthetized stuff is fine for 90% of the code using getProps().

Reply:
> dunno if others than Mosaic items do

No others do it, and I've missed the Mosaic does.
I thought about `has_meta` but (missing Mosaic) didn't invent where to use it.

Reply:
But: for documents without metadata (txt, html) `cre:getDocumentProps()` returns file name without extension as `title`, i.e. it is already synthetized.

Reply:
Not for me (?):
![image](https://github.com/koreader/koreader/assets/24273478/69f0f375-2d2f-4cb8-b8e6-30fa3f4d7340)

(I think crengine may synthetize a title from what's in the `<html><head><title>` - which is a title :) - or depending on the first lines of a text files.)

Reply:
But indeed with a .txt file (that I/we don't really care about :):
![image](https://github.com/koreader/koreader/assets/24273478/9548d2ca-ab69-4fe0-b26e-26fcce63584b)
![image](https://github.com/koreader/koreader/assets/24273478/f23b546f-062b-4a4c-ba29-e05e5d0b7aeb)

Reply:
To summarize.

(1) We cannot set `has_meta` in `getProps()` because crengine can fill the `title` for documents without metadata.

(2) In some cases (depending on the document tags) crengine does return empty `title`.
To distinguish `title` synthetized by us and by crengine we must have (and save to sdr) additional field in `props`.
What purposes this field is going to be used for?
It doesn't guarantee that the document has or has not original metadata.
It just indicates a special (rare) case of empty `title`.
My opinion it is not needed.

Reply:
Also, in Book information, where we will see the filename, and where we may want to see the metadata as-is, you may want to still show `Title: N/A`.
So, either that flag (may be a better name: `title_from_filename=true`), or store it as-is (nil or empty string), and whereever we need a non-empty title (90% of the code using it), use/add a `book_props:getTitle()`, or just store the non-empty one as `book_props.xtitle` (or something nicer).